### PR TITLE
Refactor some code

### PR DESCRIPTION
**Pull request by lokeshh**
I had a look at some places where code can be improved while documenting the functions. This PR is for that purpose.

### CODE REVIEW

File: lib/statsample-glm/glm/base.rb [hunk 1]
```diff
@@ -25,11 +25,9 @@ def initialize ds, y, opts={}
...
 
-        # TODO: Remove this const_get jugaad after 1.9.3 support is removed.
-
-        @regression = Kernel.const_get("Statsample").const_get("GLM")
-                            .const_get("#{algorithm}").const_get("#{method}")
-                            .new(@data_set, @dependent, @opts)
+        @regression = ['Statsample', 'GLM', algorithm, method]
+                      .reduce(Module, :const_get)
+                      .new @data_set, @dependent, @opts
       end
```
Comment (lokeshh) [inline lib/statsample-glm/glm/base.rb]: @v0dro I see your comment to remove const_get. Is there a way to do it without using const_get?

Comment (v0dro) [inline lib/statsample-glm/glm/base.rb]: http://stackoverflow.com/a/24420705/3992369

Comment (v0dro) [inline lib/statsample-glm/glm/base.rb]: Remove the need of chaining basically.

Comment (lokeshh) [inline lib/statsample-glm/glm/base.rb]: Is the current change fine or should I change to something like `Object.const_get('Daru::Vector')`?

Comment (v0dro) [inline lib/statsample-glm/glm/base.rb]: Keep a single `const_get`. In other words go with the SO answer.

**Comment by lokeshh** [comment]
Changed `degree_of_freedom` to `degrees_of_freedom` and `standard_error` to `standard_errors`. Also added aliases so the old names could be used. I find the tests for these functions are missing.

Should I add the tests? Also should I print a error message to tell that the old names will be removed?

**Comment by v0dro** [comment]
Aren't [these](https://github.com/SciRuby/statsample-glm/pull/28/files#diff-a44833ce44ccee8c9a9e6416a838547eR65) the tests you're looking for? Yes print an error message into $stderr.

**Comment by agisga** [comment]
I saw too that not every method is tested for every model (logistic, probit, normal etc.), and in general there is a lot of room for improvement of the tests. For example `#standard_errors` is only tested for logistic regression, and I think you are right that `#degrees_of_freedom` is never tested.
Though maybe addition of missing tests should be a separate commit (to make the commit history more organized)?

**Comment by v0dro** [comment]
Ah yes there's no tests but those for logit. And new tests should be a separate commit.

**Comment by lokeshh** [comment]
I think this is ready to be merged now. Here are the changes:
- Replace multiple `const_get`with single `const_get`.
- Deprecate `degree_of_freedom` and introduce `degrees_of_freedom`
- Deprecate `standard_error` and introduce `standard_errors`

**Pull request merged**